rest/blip_client_test.go

```diff
@@ -34,16 +34,31 @@ type BlipTesterClientOpts struct {
 	Channels               []string
 	SendRevocations        bool
 	SupportedBLIPProtocols []string
+	Collections            []string
 
 	// a deltaSrc rev ID for which to reject a delta
-	rejectDeltasForSrcRev string //nolint: structcheck // false structcheck positive due to https://github.com/golangci/golangci-lint/issues/826
+	rejectDeltasForSrcRev string // nolint: structcheck // false structcheck positive due to https://github.com/golangci/golangci-lint/issues/826
 }
 
 // BlipTesterClient is a fully fledged client to emulate CBL behaviour on both push and pull replications through methods on this type.
 type BlipTesterClient struct {
 	BlipTesterClientOpts
 
-	rt *RestTester
+	rt              *RestTester
+	pullReplication *BlipTesterReplicator // SG -> CBL replications
+	pushReplication *BlipTesterReplicator // CBL -> SG replications
+
+	CollectionClients map[string]*BlipTesterCollectionClient
+}
+
+type BlipTesterCollectionClient struct {
+	*BlipTesterClientOpts                       // Copy of parents
+	pullReplication       *BlipTesterReplicator // Pointer to parents
+	pushReplication       *BlipTesterReplicator // Pointer to parents
+	rt                    *RestTester           // Pointer to parents to allow test failures
```

Comment:
Can these not all be accessed via a pointer to the parent, instead of pointers to everything in the parent?

```suggestion
	parent *BlipTesterClient
```

Reply:
Yeah thats fair, originally didn't have so many items pointing to parent.